pkg/xray/watches.go

```diff
@@ -0,0 +1,267 @@
+	AssignedPolicies []WatchAssignedPolicy `json:"assigned_policies"`
+	WatchRecipients  []string              `json:"watch_recipients"`
+}
+
+func unpackWatch(d *schema.ResourceData) Watch {
+	watch := Watch{}
+
+	gd := WatchGeneralData{
+		Name: d.Get("name").(string),
+	}
+	if v, ok := d.GetOk("description"); ok {
+		gd.Description = v.(string)
+	}
+	if v, ok := d.GetOk("active"); ok {
+		gd.Active = v.(bool)
+	}
+	watch.GeneralData = gd
+
+	pr := WatchProjectResources{}
+	if v, ok := d.GetOk("watch_resource"); ok {
+		r := []WatchProjectResource{}
+		for _, res := range v.(*schema.Set).List() {
+			r = append(r, unpackProjectResource(res))
+		}
+		pr.Resources = r
+	}
+	watch.ProjectResources = pr
+
+	ap := []WatchAssignedPolicy{}
+	if v, ok := d.GetOk("assigned_policy"); ok {
+		policies := v.(*schema.Set).List()
+		for _, pol := range policies {
+			ap = append(ap, unpackAssignedPolicy(pol))
+		}
+	}
+	watch.AssignedPolicies = ap
+
+	var watchRecipients []string
+
+	if v, ok := d.GetOk("watch_recipients"); ok {
+		recipients := v.(*schema.Set).List()
+		for _, watchRec := range recipients {
+			watchRecipients = append(watchRecipients, watchRec.(string))
+		}
+	}
+	watch.WatchRecipients = watchRecipients
+
+	return watch
+}
+
+func unpackProjectResource(rawCfg interface{}) WatchProjectResource {
+	resource := new(WatchProjectResource)
+
+	cfg := rawCfg.(map[string]interface{})
+	resource.Type = cfg["type"].(string)
+
+	if v, ok := cfg["bin_mgr_id"]; ok {
+		resource.BinaryManagerId = v.(string)
+	}
+	if v, ok := cfg["name"]; ok {
+		resource.Name = v.(string)
+	}
+
+	if v, ok := cfg["filter"]; ok {
+		resourceFilters := unpackFilters(v.([]interface{}))
+		resource.Filters = resourceFilters
+	}
+
+	return *resource
+}
+
+func unpackFilters(list []interface{}) []WatchFilter {
+	var filters []WatchFilter
+
+	for _, raw := range list {
+		filter := new(WatchFilter)
+		f := raw.(map[string]interface{})
+		filter.Type = f["type"].(string)
+		filter.Value = f["value"].(string)
+		filters = append(filters, *filter)
+	}
+
+	return filters
+}
+
+func unpackAssignedPolicy(rawCfg interface{}) WatchAssignedPolicy {
+	policy := new(WatchAssignedPolicy)
+
+	cfg := rawCfg.(map[string]interface{})
+	policy.Name = cfg["name"].(string)
+	policy.Type = cfg["type"].(string)
+
+	return *policy
+}
+
+func packProjectResources(resources WatchProjectResources) []interface{} {
+	if len(resources.Resources) == 0 {
+		return []interface{}{}
+	}
+
+	var list []interface{}
+	for _, res := range resources.Resources {
+		resourceMap := map[string]interface{}{}
+		resourceMap["type"] = res.Type
+		if len(res.Name) > 0 {
+			resourceMap["name"] = res.Name
+		}
+		if len(res.BinaryManagerId) > 0 {
+			resourceMap["bin_mgr_id"] = res.BinaryManagerId
+		}
+		resourceMap["filter"] = packFilters(res.Filters)
+		list = append(list, resourceMap)
+	}
+
+	return list
+}
+
+func packFilters(filters []WatchFilter) []interface{} {
+	if filters == nil {
```

Comment:
I think this check is unnecessary too.

In case you wonder about nil slice. See https://stackoverflow.com/a/44305910

Reply:
yes, thanks!